### ISSUE TITLE
libero results not good

### ISSUE THREAD

**Issue by bailvwangzi** [OpenDriveLab/UniVLA#7]
Fineture libero-spatial , the train_loss can converge to 0.06, latent_action_accuracy=1, but when test with run_libero_eval.py the SR < 5%. Do you have any suggestions?

One more thing, there are some typo bugs, 

- [no _slow_output_ variable](https://github.com/OpenDriveLab/UniVLA/blob/main/vla-scripts/finetune_libero.py#L81)
- [training_phase or training_Phase](https://github.com/OpenDriveLab/UniVLA/blob/main/vla-scripts/finetune_libero.py#L283)
- [calvin_dataset not ready but import now](https://github.com/OpenDriveLab/UniVLA/blob/main/prismatic/vla/datasets/__init__.py#L4)

**Comment by swstbecrpgmail** [comment]
![Image](https://github.com/user-attachments/assets/0742bff8-a55b-420f-aebd-42e7e733c477)
![Image](https://github.com/user-attachments/assets/fad1e2d8-2bc5-4773-b1dd-b32c5a4a8f02)
Have you encountered the following error when running "torchrun --standalone --nnodes 1 --nproc-per-node 3 finetune_libero.py --dataset_name 'libero_10_no_noops' --run_root_dir 'libero_log"? My devices are three A6000 GPUs.
Thank you!

**Comment by retsuh-bqw** [comment]
@bailvwangzi Thanks a lot for testing our code and report those issues! I've fixed them in the latest commit.

Regarding the reproduction issue on LIBERO-spatial, I would first recommend verifying the dataset you're using. When training on all LIBERO suites, UniVLA's latent action accuracy typically converges around 0.7–0.8 and is unlikely to reach 1.0. Additionally, the train_loss usually stays within the range of 0.3–0.4.

**Comment by bailvwangzi** [comment]
I check again, i use the correct dataset of LIBERO. [This issue](https://github.com/OpenDriveLab/UniVLA/issues/1#issuecomment-2872749587) you said train_loss converges to below 0.04 (i think the answer above you mean 0.03-0.04, not 0.3-0.4). Do you have any other suggestions? The wandb visualization: 

![Image](https://github.com/user-attachments/assets/fbf66c3e-e100-400f-b080-0ceafd2c6e6e)

**Comment by retsuh-bqw** [comment]
The training curve seems quite unusual. Could you please provide detailed training setups? Also, are you using the correct latent action model and pretrained UniVLA checkpoints we provided?

My training curve on libero-spatial is like (train_loss = action_loss + next_latent_action_token_prediction_loss):

<img width="1997" alt="Image" src="https://github.com/user-attachments/assets/2873a07a-2e48-4ef7-865f-8681bd01689a" />

**Comment by bailvwangzi** [comment]
![Image](https://github.com/user-attachments/assets/8d0de459-bff6-4cd4-8871-f7eb76048b4f)

I use 3090 GPU * 2, batchsize=1, with lower learning_rate and large max_steps.

**Comment by bailvwangzi** [comment]
> ![Image](https://github.com/user-attachments/assets/8d0de459-bff6-4cd4-8871-f7eb76048b4f)
> 
> I use 3090 GPU * 2, batchsize=1, with lower learning_rate and large max_steps.

All the changes are shown in the picture above. The vla_path ,lam_path, data_root_dir are right? @retsuh-bqw

**Comment by retsuh-bqw** [comment]
The current batch size (GPUs $\times$ batch_size $\times$  grad_accumulation_steps = 4) is significantly smaller than our experimental setting (128). To address this, you could either: 1. Simply use more GPUs, or 2. Increase the `grad_accumulation_steps` to achieve a larger effective batch size.

**Comment by bailvwangzi** [comment]
> The current batch size (GPUs × batch_size × grad_accumulation_steps = 4) is significantly smaller than our experimental setting (128). To address this, you could either: 1. Simply use more GPUs, or 2. Increase the `grad_accumulation_steps` to achieve a larger effective batch size.

yes, i also try to use large grad_accumulation_steps=32, but  curve also not same with you.

![Image](https://github.com/user-attachments/assets/3981b2ce-c443-4784-9b1e-30535193d78a)

**Comment by retsuh-bqw** [comment]
I replicated your training configuration but noticed that the latent action accuracy exhibits significant fluctuations rather than converging quickly to 1.0. I also verified the integrity of the uploaded checkpoints via checksum, confirming they are correct. I've also tried to reproduce your results with 1 GPU but unfortunately failed.

<img width="627" alt="Image" src="https://github.com/user-attachments/assets/fa4edd53-9c01-47d1-8191-c0b7e8c20354" />

**Comment by bailvwangzi** [comment]
The error loss curve is because i use wrong **transformers** version. `pip install transformers==4.40.1` is right. Thanks again.

**Comment by Keyustar** [comment]
So it's normal that a 24GB 4090 can't handle DDP training with the default configuration, is that correct? I tried replacing DDP with DeepSpeed for training, and it works under Zero Stage 2, but the training results are not ideal. I used batch_size=1, grad_accumulation_steps=1, 8 GPUs, and max_steps=60,000—is this due to the effective batch size being too small?

> > [@bailvwangzi](https://github.com/bailvwangzi) Hello, may I ask how much VRAM your GPU has? Can it perform training under default settings without Out Of memery? I'm using 8 * 4090 GPUs, 和 it always encounters OOM with the default settings.
> 
> The full model is loaded onto each GPU. I use RTX5880 ada with batch_size=1, which consumes approximately 32GB of VRAM.

So it's normal that a 24GB 4090 can't handle DDP training with the default configuration, is that correct? I tried replacing DDP with DeepSpeed for training, and it works under Zero Stage 2, but the training results are not ideal. I used batch_size=1, grad_accumulation_steps=1, 8 GPUs, and max_steps=60,000—is this due to the effective batch size being too small?
